src/main/java/org/jabref/logic/importer/CitationBasedParserFetcher.java

```diff
@@ -0,0 +1,111 @@
+package org.jabref.logic.importer;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.net.SocketException;
+import java.net.URL;
+import java.net.UnknownHostException;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Objects;
+
+import org.jabref.logic.importer.fetcher.OpenCitationFetcher;
+import org.jabref.model.entry.BibEntry;
+import org.jabref.model.entry.field.StandardField;
+
+import kong.unirest.json.JSONArray;
+import kong.unirest.json.JSONException;
+import kong.unirest.json.JSONObject;
+
+/**
+ * Provides a convenient interface for citation-based fetcher, which follow the usual four-step procedure:
+ * 1. Send request with doi of the entry
+ * 2. Parse the response to get a list of DOI's
+ * 3. Send new request to get bibliographic information for every DOI
+ * 4. Parse the response and create a List of {@link BibEntry}
+ */
+public interface CitationBasedParserFetcher extends CitationFetcher {
+
+    /**
+     * Constructs a URL-String based on the given DOI's.
+     *
+     * @param dois the dois to look information for
+     */
+    String getURLForEntries(String... dois);
+
+    default List<BibEntry> searchCitedBy(BibEntry entry) throws FetcherException {
+        Objects.requireNonNull(entry);
+        return performSearch(entry, OpenCitationFetcher.SearchType.CITEDBY);
+    }
+
+    default List<BibEntry> searchCiting(BibEntry entry) throws FetcherException {
+        Objects.requireNonNull(entry);
+        return performSearch(entry, OpenCitationFetcher.SearchType.CITING);
+    }
+
+    /**
+     * Searches specified URL for related articles based on a {@link BibEntry}
+     * and the citation type {@link org.jabref.logic.importer.fetcher.OpenCitationFetcher.SearchType}
+     *
+     * @param entry      entry to search bibliographic information for
+     * @param searchType type of search to perform (CITING, CITEDBY)
+     * @return a list of {@link BibEntry}, which are matched by the query (may be empty)
+     * @throws FetcherException Error message passed to {@link org.jabref.gui.entryeditor.CitationRelationsTab}
+     */
+    default List<BibEntry> performSearch(BibEntry entry, OpenCitationFetcher.SearchType searchType) throws FetcherException {
```

Comment:
For a default implementaiton this feels a bit to specific. What do you think @tobiasdiez

Reply:
I agree! It should be similar to the other parser based fetcher: get the url (using an abstract method `getSearchUrl(entry, searchType)`), fetch the result using `UrlDownload` and parse it using a parser returned by an abstract method `getParser()`. Plus some generic error handling. You can essentially copy paste the existing code for a parser-based fetcher interface, and then slightly change it for the current context.

Reply:
I changed the implementation to be similar to the other fetchers using UrlDownload and getParser().